contracts/exchange-wrappers/ZeroExExchangeWrapper.sol

```diff
@@ -13,38 +13,104 @@
+ */
+contract ZeroExExchangeWrapper is I_ExchangeWrapper {
+    using SafeERC20 for IERC20;
+
+    // ============ State Variables ============
+
+    IERC20 immutable USDC_ADDRESS;
+
+    uint256 immutable USDC_ASSET_TYPE;
+
+    I_StarkwareContract public immutable STARKWARE_CONTRACT;
+
+    // ============ Constructor ============
+
+    constructor(
+        I_StarkwareContract starkwareContractAddress,
+        IERC20 usdcAddress,
+        uint256 usdcAssetType
+    )
+    {
+        STARKWARE_CONTRACT = starkwareContractAddress;
+        USDC_ADDRESS = usdcAddress;
+        USDC_ASSET_TYPE = usdcAssetType;
+
+        // Set the allowance to the highest possible value.
+        usdcAddress.safeApprove(address(starkwareContractAddress), type(uint256).max);
+    }
 
-import { I_ExchangeWrapper } from "../external/I_ExchangeWrapper.sol";
-import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 
-contract ZeroExExchangeWrapper is I_ExchangeWrapper  {
     // ============ Public Functions ============
 
     /**
-     * Exchange some amount of takerToken for makerToken. Just for tests
+    * Approve an exchange to swap an asset
+    *
+    * @param exchange Address of exchange that will be swapping a token
+    * @param token    Address of token that will be swapped by the exchange
+    */
+    function approveSwap(
+      address exchange,
+      IERC20 token
+    )
+      external
+    {
+      // safeApprove requires unsetting the allowance first.
+      token.safeApprove(exchange, 0);
+      token.safeApprove(exchange, type(uint256).max);
+    }
+
+    /**
+     * Exchange some amount of takerToken for USDC.
      *
-     * @param  tradeOriginator      Address of the initiator of the trade (however, this value
-     *                              cannot always be trusted as it is set at the discretion of the
-     *                              msg.sender)
-     * @param  receiver             Address to set allowance on once the trade has completed
-     * @param  makerToken           Address of makerToken, the token to receive
-     * @param  takerToken           Address of takerToken, the token to pay
-     * @param  inputTokenAmount     Amount of makerToken
-     * @param  orderData            Arbitrary bytes data for any information to pass to the exchange
-     * @return                      The amount of makerToken received
+
+     * @param  takerToken           Address of takerToken, the token to pay.
+     * @param  requestedFillAmount  Amount of takerToken being paid.
+     * @param  starkKey             The starkKey of the L2 account to deposit into.
+     * @param  positionId           The positionId of the L2 account to deposit into.
+     * @param  exchange             The exchange being used to swap the taker token for USDC.
+     * @param  orderData            Arbitrary bytes data for any information to pass to the exchange.
+     * @return                      The amount of makerToken received.
```

Comment:
nit: `The amount of USDC deposited to the StarkEx contract` (or something similar)